rfcs/convergence/authoring-stories.md

```diff
@@ -267,13 +270,40 @@ TBA
 
 TBA
 
-### 7. how structure stories in the storybook nav tree
+### 7. dissecting big story files into smaller ones
 
-TBA
+Components with complex API require many long stories, both for documentation and for e2e testing. This creates long files with hundres of lines, which are hard to maintain. How can these files be dissected into smaller ones?
+
+**Proposal:**
 
-### 8. dissecting big story files into smaller ones
+1. Every component can have at most one `.stories.tsx` file. This file has a default export with Storybook configuration for the docs page.
+2. If putting all stories into one file would make it too long, individual stories might be put into `.story.tsx` files as a named export, and then re-exported from `.stories.tsx` file like this: `export * from ‘./IndividualStoryFile.story’;`
+3. If individual story files are employed, `.stories.tsx` file must not contain any stories.
 
-TBA
+### 8. location and naming convention
+
+1. Story files should be in the same folder as the component which they are targeting.
+2. Name of every .stories.tsx / .story.tsx file should start with name of a component which it is targeting, to improve colocation in alphabetical file ordering. If appropriate, story name should be adjusted via configuration to a well readable name, suitable for documentation, like this:
+
+```tsx
+export const ButtonPrimary = (props: ButtonProps) => <Button {...props}>Text</Button>;
+ButtonPrimary.args = {
+  primary: true,
+};
+ButtonPrimary.storyName = 'Better story name';
+```
+
+### 9. default story
+
+Every documented component must have a story called `Default`. This story must support manipulation with [Controls](https://storybook.js.org/docs/react/essentials/controls). Ordering of stories will be set up, so this would be the first story in Storybook nav tree.
+
+### 10. design of stories
+
+Public stories should follow Fluent Design Language to give developers better feel for patterns they should utilise. For example, when a button is necessary do demonstrate usage of a component, Fluent UI Button should be used instead of a pure HTML button.
```

Comment:
Can you check if eslint will scream if you import `react-button` inside a file in `react-accordion` ?

If it's the case, we should explicitly state that having external dependencies in stories aren't in package.json should be allwoed

Reply:
Checked, it doesn’t scream :)